### ISSUE TITLE
Remaining Work for GSoC

### ISSUE THREAD

**Issue by IshaGupta18** [publiclab/simple-data-grapher#26]
Here's a checklist of what is left to do:

- [ ] Integration with plots2

Model and Controller are ready. We need to establish routes and save files against the user.

- [ ] Displaying per-user data

Once the integration with plots2 is complete, we can move to its page design and implementation. 

- [ ]  Create charts from previously uploaded files

After integration, just have to list out the files uploaded by the user and normal flow resumes

- [ ] Publish as a research note (Post Integration)

- [ ] Browsable Time Slider (Work in progress)

- [ ] Add Graph feature from the same CSV (Work in progress)

- [ ] Add multiple graphs from multiple CSVs (Work in progress)

- [ ] Export Options (Discussion)

- [ ] More UI Designing (Work in progress)

@namangupta01 @jywarren what do you think is left and can be added here? Please feel free to! Thanks!

**Comment by jywarren** [comment]
Awesome, how about import options? Like, standard modules to import from
CSV URL, string, maybe even Google Spreadsheets later?

**Comment by IshaGupta18** [comment]
Yes, we can work on that! While you're at it, could you please have a look
at a couple of PRs and issues we have put up in this repo? Thanks a lot!

**Comment by IshaGupta18** [comment]
@jywarren we have importing feature from CSV URL (basically from a remote file). We can work for a big text area box for a string, and look up google spreadsheets too.

**Comment by IshaGupta18** [comment]
@jywarren as for importing data from a Google Sheet, we will need the user to sign in with the google, in order to access the user's data, so we will need a little planning around this.

If the user wants to upload a google sheet, they will simply have to sign in with a google account, and then we can access the sheet's data.

What do you think about this?

**Comment by jywarren** [comment]
Actually i believe someone in LEL figured out how to do this without a
signin - as long as the sheet is public! Take a look here -- and leave a
comment for more info?

https://github.com/publiclab/leaflet-environmental-layers/pull/97

Awesome!

**Comment by IshaGupta18** [comment]
So I tried and figured it out, and I found out, that for this to work, the sheet will have to be published on the web, something like what has been done at LEL:
Like here
![image](https://user-images.githubusercontent.com/40794215/58722183-6b7b1c80-83f4-11e9-99dd-05435292fced.png)

Then we can fetch data without logging in.

The question here is, is it worth the user, to publish data that might be sensitive to them on the web v/s asking the user to login to grant access. The latter one involves more work, however, that might be a safer option to the user. But since the user is allowing us to access the file's data, they may not mind publishing on the web, though that will involve them doing some additional work before coming and using the library. It is basically a tradeoff. What do you think @jywarren ?

**Comment by IshaGupta18** [comment]
@IgorWilbert @geekychasser what do you think about this?

**Comment by jywarren** [comment]
I think perhaps we should offer some good guidance (with screenshots) on how to make it public, but simultaneously warn people what that will mean with regard to privacy. If they don't want to do it, they could just upload instead of using the Google Spreadsheets process.

**Comment by IshaGupta18** [comment]
Yes, I think that would be apt. We can write the steps required to upload a
Google Sheet and it's repercussions. I have implemented this feature here!
